### ISSUE TITLE
Fever API support?

### ISSUE THREAD

**Issue by SkyCrawl** [SSilence/selfoss#711]
Hello,

is `selfoss` ever going to support the Fever API? Despite it already supports a JSON API, I take it that this API is not compatible with RSS clients, unless they specifically aimed to support it?

What I'm trying to do is to find a good RSS aggregator to run on my NAS, ideally using Docker, and access it from multiple devices so it would be the aggregator's job to sync everything. I've already tested that Fever API is great for this using [Coldsweat](https://github.com/passiomatic/coldsweat). Coldsweat doesn't seem to have a docker image yet though, unlike `selfoss` :).

**Comment by niol** [comment]
Can you explain what the Fever API is and provide links to doc and test suite?

**Comment by SkyCrawl** [comment]
Thanks for a fast reply @niol ,

I think [this](http://www.feedafever.com/api) is Fever API. As far as I know, it's the API of the [Fever](www.feedafever.com) service. Many clients seem to support it :).

**Comment by niol** [comment]
I still do not understand what features it would bring to selfoss. Maybe someone else will be interested.

**Comment by aminecmi** [comment]
@niol it would be easier to use selfoss with other mobile apps that use fever api.
That could be usefull.

**Comment by rajo** [comment]
Yep, I'd like such a feature too: currently I'm using ttrss on my server, but I would like to move away from it and selfoss looks like a good alternative. Nevertheless I'm using reeder on my iphone, which currently connects to ttrss via the fever api.

**Comment by matthiasballreich** [comment]
i would like the fever api integration, too.

**Comment by jtojnar** [comment]
I do not think adding a second API is a good idea – maintaining two separate APIs would be bothersome. Replacing the current one with some other, more widely implemented API is probably a better idea. I am not sure if the new API should be Fever API though – there are probably more commonly used alternatives.

* Few years back there was an [attempt](https://rss-sync.github.io/Open-Reader-API/rssconsensus/) to adopt a universal protocol, essentially it is copy of the Google Reader API so there will be many clients (possibly not developed anymore though).
* Feedly has its own API and boasts that [over 200 apps](https://feedly.com/apps.html) support it
* Then there are [FeedBin](https://feedbin.com/apps) and [Feed Wrangler](https://feedwrangler.net/welcome.html) but I have never heard of them and they are probably even more obscure than Fever.

**Comment by timbroder** [comment]
I currently use Feedbin and am looking to move to self hosted (I want longer archiving times)

In the apps I've used: Reeder, Unread, and Readkit, all have Fever API support. It seems to be the only self-hosted API with any traction. I've love for selfoss to support it.

If you moved to the feedly API, you'd have to get the different app developers to let you put your own domain in. The apps that have fever support already allow that

An example of a tool piggy-backing on that API definition is https://github.com/pietheinstrengholt/rssmonster

**Comment by amaccuish** [comment]
I've nearly finished a script that provides the fever api for selfoss. It's a bit ugly, calls the selfoss json api and rewrites requests and responses but gets the job done :)

**Comment by kosciak9** [comment]
Is it on a roadmap? Or simply done when someone pull requests it? :)

**Comment by amaccuish** [comment]
I've updated my script so it works a lot faster. I'll work over the coming weeks to directly use the api and prep something worth a pr :)

**Comment by kosciak9** [comment]
Oh nice! Cool then, waiting patiently. :)

**Comment by jtojnar** [comment]
I will not merge this, our API is already pretty gruesome. However, I would like to make possible to implement this as a third party plug-in. Could you comment on #877.
